webapp/votes.py

```diff
@@ -45,27 +47,21 @@ def vote():
         token = session.get("token")
         if not token:
             return redirect("/fail")
-        try:
-            conn = http.client.HTTPSConnection(api)
-            headers = {"Authorization":"Bearer " + token}
-            conn.request("GET","/players/me",headers=headers)
-            response = conn.getresponse()
-            data = response.read()
-            vid = json.loads(data)["data"]["id"]
-        except Exception as e:
-            print(e)
-            return redirect("/fail")
+        req = urllib.request.Request(api + "/players/me", method="GET",
+            headers={"Authorization":"Bearer " + token})
+        with urllib.request.urlopen(req) as response:
+            vid = json.loads(response.read().decode('utf8'))["data"]["id"]
 
         candidate = request.form.get("candidate")
         if not candidate:
             return redirect("/fail")
 
         conn = pymysql.connect(**db_creds)
-        with conn as cursor:
-            cursor.execute("DELETE FROM votes WHERE vid=?",(vid,))
-            cursor.execute("INSERT INTO votes VALUES(?,?)",(vid,candidate))
+        with conn.cursor() as cursor:
+            cursor.execute("DELETE FROM votes WHERE vid=%s",(vid))
```

Comment:
`(vid,)` and `(vid)` are not the same. The first one is a tuple with one value, while the second one is a value within unnecessary parentheses - so PyCharm will eventually remove them.

Technically it doesn't matter because the result is the same, but as soon as someone wants to add another parameter he might be tempted to do it as two arguments instead of within a tuple, which will then not work as expected.